packages/flutter/lib/src/material/navigation_rail.dart

```diff
@@ -0,0 +1,761 @@
+  final bool extended;
+
+  /// The leading widget in the rail that is placed above the destinations.
+  ///
+  /// This is commonly a [FloatingActionButton], but may also be a non-button,
+  /// such as a logo.
+  final Widget leading;
+
+  /// The trailing widget in the rail that is placed below the destinations.
+  ///
+  /// This is commonly a list of additional options or destinations that is
+  /// usually only rendered when [extended] is true.
+  final Widget trailing;
+
+  /// Defines the appearance of the button items that are arrayed within the
+  /// navigation rail.
+  final List<NavigationRailDestination> destinations;
+
+  /// The index into [destinations] for the current active
+  /// [NavigationRailDestination].
+  final int currentIndex;
+
+  /// Called when one of the [destinations] is selected.
+  ///
+  /// The stateful widget that creates the navigation rail needs to keep
+  /// track of the index of the selected [NavigationRailDestination] and call
+  /// `setState` to rebuild the navigation rail with the new [currentIndex].
+  final ValueChanged<int> onDestinationSelected;
+
+  /// The elevation for the inner side of the rail.
+  ///
+  /// The shadow only shows on the inner side of the rail.
+  ///
+  /// In LTR configurations, the inner side is the right side, and in RTL
+  /// configurations, it is the left side.
+  final double elevation;
+
+  /// The alignment for the [NavigationRailDestination]s as they are positioned
+  /// within the [NavigationRail].
+  ///
+  /// Navigation rail destinations can be aligned as a group to the [top],
+  /// [bottom], or [center] of a layout.
+  final NavigationRailGroupAlignment groupAlignment;
+
+  /// Defines the layout and behavior of the labels for the default, unextended
+  /// [NavigationRail].
+  ///
+  /// When the navigation rail is extended, the labels are always shown.
+  ///
+  /// See also:
+  ///
+  ///   * [NavigationRailLabelType] for information on the meaning of different
+  ///   types.
+  final NavigationRailLabelType labelType;
+
+  /// The [TextStyle] of the unselected [NavigationRailDestination] labels.
+  ///
+  /// When the [NavigationRailDestination] is selected, the
+  /// [selectedLabelTextStyle] will be used instead.
+  final TextStyle unselectedLabelTextStyle;
+
+  /// The [TextStyle] of the [NavigationRailDestination] labels when they are
+  /// selected.
+  ///
+  /// When the [NavigationRailDestination] is not selected,
+  /// [unselectedLabelTextStyle] will be used.
+  final TextStyle selectedLabelTextStyle;
+
+  /// The default size, opacity, and color of the icon in the
+  /// [NavigationRailDestination].
+  ///
+  /// If this field is not provided, or provided with any null properties, then
+  /// a copy of the [IconThemeData.fallback] with a custom [NavigationRail]
+  /// specific color will be used.
+  final IconThemeData unselectedIconTheme;
+
+  /// The size, opacity, and color of the icon in the selected
```

Comment:
This isn't quite exactly right (likewise for the text style parameters) because it's about the appearance of icons in the destination's icon ... but OK.

Reply:
Updated these docs